BringYourOwnBoard.md

```diff
@@ -0,0 +1,157 @@
+
+1. adu-swupdate-hw-compat
+2. adu-device-info-files
+3. adu-agent-service
+
+If you have questions about what these are and what they do in the actual image please read the doc in [meta-azure-device-update here]( https://github.com/Azure/meta-azure-device-update). 
+
+## recipes-extended
+
+### What does it do? 
+
+The [recipes-extended](./recipes-extended/) section of the repository is the tooling for generating the SwUpdate file (the `*.swu`) that allows SwUpdate to install the image. Each `*.swu` file must contain a [sw-description file](./recipes-extended/images/adu-update-image/raspberrypi4-64/sw-description) which describes the goal state of the device. You can read more about what a sw-description file is, options you can use, and how to write your own [here](https://sbabic.github.io/swupdate/sw-description.html). 
+
+As apart of the `*.swu` generation swupdate will require that you sign the file with a private key so that the public key that is installed on the device at build time can be used to verify the image. You can read more about how we generate our keys for the proof of concept [here](https://github.com/Azure/iot-hub-device-update-yocto/blob/main/keys/README.md) as well as some recommendations for what to look for in production. 
+
+### What do I need to think about? 
+
+For a customer looking at porting this section over there's not a lot you need to change. You need to edit the file such that it depends on your own custom image's bitbake file, edit the sw-description file to reflect what you would like the system to look like after an update, and provide tooling for accessing a private signing key. 
+
+## recipes-graphics
+
+### What does it do? 
+This just sets up graphics for the RPi. 
+### What do I need to think about? 
+There's nothing that affects customers in this section. You should just be able to entirely omit it from your port. 
+
+## recipes-support
+### What does it do? 
+
+In our project the [recipes-support](./recipes-support/) directory appends some of the extra tooling around swupdate. We use the [adu-swupdate-hw-compat.bb](./recipes-support/adu-swupdate-hw-compat/adu-swupdate-hw-compat.bb) to install a hardware compatibility file onto the image which swupdate looks for when completing an update. The version held within this file MUST match the one inside of the `sw-description` file mentioned above. 
+
+We also append swupdate with our own configuration inside of the [defconfig file](./recipes-support/swupdate/swupdate/raspberrypi4-64/defconfig). 
+### What do I need to think about? 
+
+For your `hw-compat` file that SwUpdates uses you should have a method for using either a hash of the hardware compatibilities, use a model number for the device you're deploying updates to, or really anything other than just a number that does not allow for any sort of granularity. The version held within this file MUST mach the one inside of the `sw-description` file mentioned above. 
```

Comment:
For this: "for using either a hash of the hardware compatibilities, use a model number for the device you're deploying updates to, or really anything other"

1. "using" should be used (ha!) in both cases - so "for using either..." and "using a model number...".
2. "either" means you have two options, but you actually have a third (anything other). Just deleting "either" works fine, I think.